src/main/kotlin/cacophony/codegen/instructions/matching/InstructionMatcher.kt

```diff
@@ -125,6 +128,13 @@ class InstructionMatcherImpl(
             for ((nestedNode, nestedPattern) in node.children().zip(pattern.children())) {
                 if (!tryMatch(nestedNode, nestedPattern, matchMetadata)) return false
             }
+
+            if (node is CFGNode.Call) {
```

Comment:
I am personally in favor of creating a new slot type. It should not significantly increase the amount of code, while preserving consistency in the implementation of patterns.